invenio_rdm_records/cli.py

```diff
@@ -278,3 +280,176 @@ def rebuild_index():
     affs_service.rebuild_index(identity=system_identity)
 
     click.secho("Reindexed records and vocabularies!", fg="green")
+
+
+# CUSTOM FIELDS
+
+
+@click.group()
+def custom_fields():
+    """InvenioRDM custom fields commands."""
+
+
+@custom_fields.group()
+def records():
+    """InvenioRDM custom fields records commands."""
+
+
+# helper functions
+def _prepare_mapping(fields_names, available_fields):
+    """Prepare ES mapping properties for each field."""
+    available_fields = {field.name: field for field in available_fields}
+    fields = []
+    if fields_names:
+        for field_name in fields_names:
+            fields.append(available_fields.get(field_name))
+    else:
+        fields = available_fields.values()
+
+    properties = {}
+    for field in fields:
+        properties[f"custom_fields.{field.name}"] = field.mapping
+
+    return properties
+
+
+def _exists(field_name, index):
+    """Check if a field exists in `index`'s mapping."""
+    mapping = list(index.get_mapping().values())[0]["mappings"]
+
+    parts = field_name.split(".")
+    for part in parts:
+        mapping = mapping["properties"]  # here to avoid last field access
+        if part not in mapping.keys():
+            return False
+        mapping = mapping[part]
+
+    return True
+
+
+@records.command("create")
```

Comment:
Silly question: given that we have `index init`, shall we use `init` cmd name instead?

Reply:
Agree, `invenio custom-fields init`, there is also `index create` but is to create a full index. Adding templates or mappings happens in `init` so it also makes more sense to call it `init` here